wp-pgp-encrypted-emails.php

```diff
@@ -937,6 +1272,44 @@ private static function prepareMail ($to, $subject, $message, $headers, $attachm
                 ));
             }
         }
+        else if ($pub_cert) {
+            try {
+                // Headers for encrypted part
+                $message = $headers."\n\n".$message;
+
+                // Set up files for openssl pkcs7 function, based on Elwing's S/MIME plugin
+            		$clearfile = tempnam("temp","email");
+            		$encfile = $clearfile.".enc";
+            		$clearfile .= ".txt";
+            		$fp = fopen($clearfile,"w");
+            		fwrite($fp,$message);
+            		fclose($fp);
+
+                // Do the encryption
+                if (openssl_pkcs7_encrypt($clearfile,$encfile,$pub_cert,array("From" => get_option('admin_email'))))
+                {
+                    $data = file_get_contents($encfile);
+                    //separate header and body, to use with mail function
+                    $parts = explode("\n\n",$data, 2);
+                    $message = $parts[1];
+                    $headers = $parts[0];
+                }
+
+            		//Unlink the temporary files:
+            		unlink($clearfile);
+            		unlink($encfile);
+
+          } catch (Exception $e) {
```

Comment:
This `catch` block fails to `unlink()` the files. If an error is thrown after or during the call to `fwrite()`, above, there will be "sensitive" data (email message contents that are supposed to be encrypted but have not yet been encrypted) on the filesystem. For this reason, I think it is wiser if you moved the declaration of `$clearfile` and `$encfile` outside the `try`/`catch` block and did the same with the calls to `unlink()`.